From 31e713e9f615c14786b7b015df8d2cf24c53841a Mon Sep 17 00:00:00 2001
From: Scott Murray <scott.murray@konsulko.com>
Date: Fri, 23 May 2025 15:30:59 -0400
Subject: [PATCH] Use pkgconfig for EGL and GLES libraries

Mali GPU platforms that do not use Mesa have EGL and GLES implemented
via libmali instead of the typical libEGL, etc.  Using pkgconfig for
the link options for those libraries allows that to work transparently.

Upstream-Status: Pending
Signed-off-by: Scott Murray <scott.murray@konsulko.com>
---
 shell/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/shell/CMakeLists.txt b/shell/CMakeLists.txt
index 7c82d0c..d8f81d2 100644
--- a/shell/CMakeLists.txt
+++ b/shell/CMakeLists.txt
@@ -88,7 +88,9 @@ target_include_directories(${PROJECT_NAME}
 )
 
 if (BUILD_BACKEND_WAYLAND_EGL)
-    target_link_libraries(${PROJECT_NAME} PRIVATE EGL GLESv2)
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(EGL egl glesv2 IMPORTED_TARGET REQUIRED)
+    target_link_libraries(${PROJECT_NAME} PRIVATE PkgConfig::EGL)
 endif ()
 if (BUILD_BACKEND_WAYLAND_VULKAN)
     target_link_libraries(${PROJECT_NAME} PRIVATE Vulkan::Headers)
-- 
2.49.0

